File: SpeedRunningLeaderboards/Models/Runner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpeedRunningLeaderboards.Models
{
	public class Runner : DiscordLogin
	{
		public Guid RunnerID { get; set; }
		public int? RegionID { get; set; }
		public Region? Region { get; set; }
		public DateTime SignUpDate { get; set; }
		public ICollection<Authority> RunnerAuthorities { get; set; } = new List<Authority>();
		public ICollection<Social> Socials { get; set; } = new List<Social>();
		public Runner()
		{

		}
		public Runner(int? regionId
								 , DateTime signUpDate
								 , ICollection<Authority> authorities
								 , ICollection<Social> socials
								 , string discordLoginID
								 , string username
								 , string discriminator
								 , string avatar
								 , bool? bot
								 , bool? system
								 , bool? mfaEnabled
								 , string? locale
								 , bool? verified
								 , string? email
								 , int? flags
								 , int? premiumType
								 , string? bannerColor
								 , int? accentColor
								 , int? publicFlags) 
					 : base(discordLoginID
								 , username
								 , discriminator
								 , avatar
								 , bot
								 , system
								 , mfaEnabled
								 , locale
								 , verified
								 , email
								 , flags
								 , premiumType
								 , bannerColor
								 , accentColor
								 , publicFlags
						)
		{
			RegionID = regionId;
			SignUpDate = signUpDate;
			RunnerAuthorities = authorities;
			Socials = socials;
		}

		public Runner(DiscordLogin discordUser, int? regionId, DateTime signUpDate, ICollection<Authority> authorities, ICollection<Social> socials) : base(discordUser)
		{
			RegionID = regionId;
			SignUpDate = signUpDate;
			RunnerAuthorities = authorities;
			Socials = socials;
		}
	}
}

File: frontend/src/Models/Runner.ts
import DiscordUser from "./DiscordUser";

export default class Runner implements DiscordUser {
	id: string;
	username: string;
	discriminator: string;
	avatar: string;
	bot?: boolean;
	system?: boolean;
	mfa_enabled?: boolean;
	locale?: string;
	verified?: string;
	email?: string;
	flags?: number;
	premium_type?: number;
	public_flags?: number;
	banner_color?: string;
	accent_color?: string;
	runnerID: string;
	regionID?: number;
	signUpDate: Date;
	constructor(discordloginid: string, 
							username: string,
							discriminator: string,
							avatar: string,
							runnerId: string,
							signUpDate: Date,
							bot?: boolean,
							system?: boolean,
							mfa_enabled?: boolean,
							locale?: string,
							verified?: string,
							email?: string,
							flags?: number,
							premium_type?: number,
							public_flags?: number,
							banner_color?: string,
							accent_color?: string,
							regionId?: number
	) {
		this.id = discordloginid;
		this.username = username;
		this.discriminator = discriminator;
		this.avatar = avatar;
		this.bot = bot;
		this.system = system;
		this.mfa_enabled = mfa_enabled;
		this.locale = locale;
		this.verified = verified;
		this.email = email;
		this.flags = flags;
		this.premium_type = premium_type;
		this.public_flags = public_flags;
		this.banner_color = banner_color;
		this.accent_color = accent_color;
		this.runnerID = runnerId;
		this.regionID = regionId;
		this.signUpDate = signUpDate;
	}
}